extensions/amp-video/0.1/amp-video.js

```diff
@@ -83,18 +90,47 @@ class AmpVideo extends AMP.BaseElement {
    * @override
    */
   preconnectCallback(opt_onLayout) {
-    const videoSrc = this.getVideoSource_();
+    const videoSrc = this.getVideoSourceForPreconnect_();
     if (videoSrc) {
       assertHttpsUrl(videoSrc, this.element);
       this.preconnect.url(videoSrc, opt_onLayout);
     }
   }
 
+  /**
+   * @override
+   *
+   * @overview
+   * AMP Cache may selectively cache certain video sources (based on size, type
```

Comment:
I would remove size given that the Cache doesn't know the video size at the time of rewriting the URL. In that case it will return a 404 at serving. Currently it may rewriting depending on other parameters (e.g., what AMP extensions are present amp-story and video type).